Java/core/src/main/java/com/amazon/randomcutforest/tree/IBoxCache.java

```diff
@@ -59,4 +59,12 @@
      */
     void addToBox(int index, Point point);
 
+    /**
+     * changes the fraction of boxes cached dynamically
+     * 
+     * @param fraction new fraction of boxes to be cached
+     */
+
+    void reInitialize(double fraction);
```

Comment:
I think we should call this `setCacheFraction`. Implementations can decide what to do with the fraction value (i.e., if reinitialization is required that's an implementation detail).

Reply:
Done